aqua/qt5/Portfile

```diff
@@ -1676,9 +1705,40 @@ foreach {module module_info} [array get modules] {
                 # do not opportunistically find MacPorts libraries (e.g. X11 libraries)
                 patchfiles-append patch-qtwebengine_tests.diff
 
-                # chromium/build/mac/find_sdk.py is written for macOS
-                # 10.* versioning; tweak to find 1[0-9].*
-                patchfiles-append patch-qtwebengine_sdk.diff
+                # run syncqt.pl to generate headers needed to build from git snapshot (rather than release tarball)
+                # system Perl is sufficient, since syncqt.pl only uses standard core modules
+                # (although future macOS may not include Perl)
+                #pre-build {
+                #    system -W ${worksrcpath} \
+                #        "/usr/bin/perl ${prefix}/libexec/${name}/bin/syncqt.pl -version ${version} ."
+                #}
+                #
+                # alternatively, apply syncqt.pl output from a patch to avoid build dependency on perl
+                post-patch {
+                    set syncqt_patch patch-qtwebengine_syncqt.pl_headers.diff
+                    set hex_version [format "0x%02X%02X%02X" {*}[split ${version} "."]]
+                    ui_info "$UI_PREFIX Applying ${syncqt_patch} (using sed)"
+                    system -W ${worksrcpath} \
+                        "/usr/bin/sed \
+                            -e 's|@@QTWEBENGINE_VERSION@@|${version}|g' \
+                            -e 's|@@QTWEBENGINE_VERSION_HEX@@|${hex_version}|g' \
+                            ${filespath}/${syncqt_patch} \
+                         | /usr/bin/patch -p0"
+                }
```

Comment:
Please let me know if you believe it is not appropriate to apply syncqt.pl output from a patch instead of running syncqt.pl (e.g. concerns over trust or bootstrapping).